ikarus/finiteelements/fehelper.hh

```diff
@@ -38,4 +38,80 @@ auto localSolutionBlockVector(const typename Traits::FERequirementType::Solution
         localX[i][j] = x[localView.index(localView.tree().child(j).localIndex(i))[0]];
   return localX;
 }
+
+/**
+ * \brief A function which loops over all the nodes of a tree and performs different actions
+ * for a power node and a leaf node depending on the corresponding functor passed.
+ *
+ * \details This function is inspired from the function Dune::TypeTree::Detail::forEachNode
+ * available in dune/typetree/traversal.hh
+ *
+ * \tparam T Type of tree.
+ * \tparam ChildIndex Type of the index of the child.
+ * \tparam PowerFunc Type of the functor called for a power node.
+ * \tparam LeafFunc Type of the functor called for a leaf node.
+ * \param tree Tree whose nodes are to be looped over.
+ * \param childIndex The index of the child.
+ * \param powerFunc A functor to be called for a power node.
+ * \param leafFunc A functor to be called for a leaf node.
+ */
+template <class T, class ChildIndex, class PowerFunc, class LeafFunc>
+void forEachLeafOrPowerNode(T&& tree, ChildIndex&& childIndex, PowerFunc&& powerFunc, LeafFunc&& leafFunc) {
```

Comment:
Why are you using this childIndex? This is not needed I suppose. Simply pass the child. Additionally the signature would be better if you just pass the `treePath` as in https://gitlab.dune-project.org/staging/dune-typetree/-/blob/master/dune/typetree/traversal.hh?ref_type=heads#L164. Why did you modify the signature?